.github/workflows/git-commit-checks.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import git
+import json
+import argparse
+
+GITHUB_WORKSPACE = os.environ.get('GITHUB_WORKSPACE')
+GITHUB_SHA       = os.environ.get('GITHUB_SHA')
+GITHUB_BASE_REF  = os.environ.get('GITHUB_BASE_REF')
+
+# Sanity check
+if GITHUB_WORKSPACE is None or GITHUB_SHA is None or GITHUB_BASE_REF is None:
+    print("Error: this script is designed to run as a Github Action")
+    exit(1)
+
+#----------------------------------------------------------------------------
+
+def print_result(good_list, bad_list, skipped_list):
+    def _print_list(msg, items, prefix=""):
+        print(msg)
+        for item in items:
+            print(f"{prefix}- {item}")
+
+    passed = True
+    if len(good_list):
+        msg = "\nThe following commits passed all tests:"
+        _print_list(msg, good_list)
+
+    if len(bad_list):
+        passed = False
+        # The "::error ::" token will cause Github to highlight these
+        # lines as errors
+        msg = f"\n::error ::The following commits caused this test to fail"
+        _print_list(msg, bad_list, prefix="::error ::")
+
+    if len(skipped_list):
+        msg = "\nThe following commits were skipped (reverts, merges):"
+        _print_list(msg, skipped_list)
+
+    return passed
+
+#----------------------------------------------------------------------------
+
+# Global regexp, because we use it every time we call
+# check_signed_off() (i.e., for each commit in this PR)
+prog_sob = re.compile(r'Signed-off-by: (.+) <(.+)>')
+
+def check_signed_off(config, repo, commit, skipped_list, bad_list):
```

Comment:
This is a really weird function signature that, frankly, hurt my brain.  We pass in the skipped and bad list, but not the good list.  Then have a return value to tell us whether to put it on the good list?  Why not just have the function return either be a "good, bad, defer, abort" OR "worked/fatal flaw" and have the good list also be an argument?  I'm sure there are other options I haven't thought of as well, but this one seems like a particularly weird choice.

Reply:
Yeah, this evolved as the requirements in my head for this script changed over time (e.g., from cherry-pick checker to "all the git commit checks").  I'll re-factor.